src/dom-parts/ChildNodePart.js

```diff
@@ -48,8 +51,25 @@ const removeNodesBetweenComments = (commentNode) => {
  * @return {Node[]}
  */
 const diffChildNodes = function (parentNode, domChildNodes, templateChildNodes, anchorNode) {
+    // We always want to have the two arrays (domChildNodes and templateChildNodes) to have the same length.
+    // We also don't ever want to replace/diff the last node of the domChildNodes as it will always be the
+    // closing comment marker (/template-part) aka the anchorNode. So we push it to the end by inserting undefined items.
+    if (templateChildNodes.length > domChildNodes.length) {
+        const nodesToFill = templateChildNodes.length - domChildNodes.length;
+        for (let i = 0; i < nodesToFill; i++) {
+            domChildNodes.splice(domChildNodes.length - 1, 0, undefined);
+        }
+    }
+
+    if (domChildNodes.length > templateChildNodes.length) {
+        const nodesToFill = domChildNodes.length - templateChildNodes.length;
+        for (let i = 0; i < nodesToFill; i++) {
+            templateChildNodes.splice(templateChildNodes.length - 1, 0, undefined);
+        }
+    }
+
+    let length = domChildNodes.length;
```

Comment:
Why do we need the assignement ?
Are domChildNodes mutated by the following code ?

I think this should be a const / or simply be used in the for loop.